session.go

```diff
@@ -57,7 +59,9 @@ func (ses *Session) Stream(ctx context.Context, in <-chan []cid.Cid) (<-chan blo
 				s.Enqueue(reqs...)
 			case <-ses.ctx.Done():
 				cancel()
-				err <- ses.err
+				if ses.err != nil {
+					err <- ses.err
```

Comment:
Hopefully the last thing: we have to be sure that this routine won't leak. This may happen if there is an error but the client has not read for some reason. Handle that case by allocating error chan buf size to 1 and routine will never block on this line hence will never leak/stuck/